CRM/Civicase/Hook/alterAngular/AngularChangeSet.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+use CRM_Civicase_ExtensionUtil as ExtensionUtil;
+use Civi\Angular\ChangeSet;
+
+/**
+ * Angular ChangeSet Helper Class.
+ */
+class CRM_Civicase_Hook_alterAngular_AngularChangeSet {
+
+  /**
+   * Returns ChangeSet for the case type category field.
+   *
+   * This ChangeSet is needed for the core Case Type create/Edit screen.
+   *
+   * @return \Civi\Angular\ChangeSet
+   *   Angular ChangeSet.
+   */
+  public static function getForCaseTypeCategoryField() {
+    $path = CRM_Core_Resources::singleton()
+      ->getPath(ExtensionUtil::LONG_NAME, 'templates/CRM/Civicase/ChangeSet/CaseTypeCategory.html');
+    $caseTypeCategoryContent = file_get_contents($path);
+
+    return ChangeSet::create('case-type-category')
+      ->alterHtml('~/crmCaseType/caseTypeDetails.html', function (phpQueryObject $doc) use ($caseTypeCategoryContent) {
+        $doc->find("div[crm-ui-field*=name: 'caseTypeDetailForm.caseTypeName']")->after($caseTypeCategoryContent);
```

Comment:
is there a way to determine if `$doc->find("div[crm-ui-field*=name: 'caseTypeDetailForm.caseTypeName']")` returned an element and if not to throw an error?

This would silently fail if the selector is not found and can lead to a regression that nobody will notice unless we pay attention to it.